File: tests/admin_login_test.js
const { Builder, By, until } = require('selenium-webdriver');

async function testAdminLogin() {
  console.log("Starting browser...");
  let driver = await new Builder().forBrowser('chrome').build();

  try {
    console.log("Navigating to admin login page...");
    await driver.get('http://localhost:3000/admin-login');
    await driver.manage().window().maximize();

    const emailField = await driver.wait(until.elementLocated(By.css("input[type='email']")), 10000);
    const passwordField = await driver.findElement(By.css("input[placeholder='••••••••']"));
    const submitBtn = await driver.findElement(By.css("button[type='submit']"));

    const testEmail = "admin@expense.com";
    const testPassword = "adminPass99";

    console.log(`Entering Admin Email: ${testEmail}`);
    await emailField.sendKeys(testEmail);

    console.log("Entering Admin Security Key...");
    await passwordField.sendKeys(testPassword);

    console.log("Clicking Access Dashboard...");
    await driver.executeScript("arguments[0].click();", submitBtn);

    await driver.wait(until.urlContains('/admin'), 5000);
    console.log("\n✅ TEST PASSED: Successfully logged in as admin and redirected to the Admin Dashboard!");

  } catch (error) {
    console.log(`\n❌ TEST FAILED: ${error.message}`);
  } finally {
    await driver.sleep(3000);
    console.log("Closing browser...");
    await driver.quit();
  }
}

testAdminLogin();

File: tests/admin_login_test.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
import time

def test_admin_login():
    print("Starting browser...")
    driver = webdriver.Chrome()
    
    try:
        # Navigate to Admin Login Page
        print("Navigating to admin login page...")
        driver.get("http://localhost:3000/admin-login")
        driver.maximize_window()
        
        wait = WebDriverWait(driver, 10)
        
        email_field = wait.until(EC.presence_of_element_located((By.CSS_SELECTOR, "input[type='email']")))
        password_field = driver.find_element(By.CSS_SELECTOR, "input[placeholder='••••••••']")
        submit_button = driver.find_element(By.CSS_SELECTOR, "button[type='submit']")
        
        # Test Credentials
        test_email = "admin@expense.com"
        test_password = "adminPass99"
        
        print(f"Entering Admin Email: {test_email}")
        email_field.send_keys(test_email)
        
        print("Entering Admin Security Key...")
        password_field.send_keys(test_password)
        
        print("Clicking Access Dashboard...")
        driver.execute_script("arguments[0].click();", submit_button)
        
        # Verify Results: Wait to see if we get routed to the admin dashboard
        wait.until(EC.url_contains("/admin"))
        print("\n✅ TEST PASSED: Successfully logged in as admin and redirected to the Admin Dashboard!")
        
    except Exception as e:
        print(f"\n❌ TEST FAILED: {str(e)}")
        
    finally:
        time.sleep(3)
        print("Closing browser...")
        driver.quit()

if __name__ == "__main__":
    test_admin_login()
